docs/http-api-integration.md

```diff
@@ -0,0 +1,112 @@
+# Complementing RESTful with JSON:API or GraphQL
+_**Specifications Beyond the Six Architectural Constraints of RESTful**_
+
+This article intends to explain why Hydrogen supports specific RESTful-styled specifications; [JSON:API](https://jsonapi.org/format/) and [GraphQL](https://graphql.org/learn/).
+
+## Prerequisite Understanding and Terms
+As outlined by [Roy Fielding](https://www.ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm) (citation), the **Six Architectural Constraints for RESTful API** are as follows:
+1. Client-Server
+2. Stateless
+3. Cache
+4. Uniform Interface
+5. Layered System
+6. Code-On-Demand
+
```

Comment:
Jonno needs to state two definitions:
RPC: Remote Procedure Control
RESTful: Representational 

Having these stated will be helpful in showing the difference in behaviour how client and server interactions between JSON:API and GraphQL (In JSON:API, the client describes the action that it wants, while in GraphQL the client is telling the server what to do)